### ISSUE TITLE
Cluster up with service catalog fails, role already exists

### ISSUE THREAD

**Issue by spadgett** [openshift/origin#15562]
Running this command

```
$ oc cluster up --service-catalog=true --version=latest --loglevel=2
```

fails with this error

```
-- Installing service catalog ...
I0731 08:29:38.294872   17526 servicecatalog.go:93] instantiating service catalog template with parameters map[SERVICE_CATALOG_SERVICE_IP:172.30.1.2 CORS_ALLOWED_ORIGIN:127.0.0.1 SERVICE_CATALOG_IMAGE:openshift/origin-service-catalog:latest]
FAIL
   Error: cannot instantiate service catalog template
   Caused By:
     Error: cannot create objects from template openshift-infra/service-catalog
     Caused By:
       Error: role.authorization.openshift.io "extension-apiserver-authentication-reader" already exists
```

##### Version

oc v3.6.0-alpha.2+4443818-1274
kubernetes v1.7.0+695f48a16f
features: Basic-Auth

@csrwng @deads2k

**Comment by deads2k** [comment]
Looks like cluster up needs to reconcile or at least check instead of creating.  This role is going to be present in 3.7 and beyond.

**Comment by bparees** [comment]
maybe i'll just name it something else :)

**Comment by csrwng** [comment]
or do the equivalent of apply

**Comment by deads2k** [comment]
> maybe i'll just name it something else :)

You shouldn't.  That could encourage people to write to the "wrong" name

**Comment by spadgett** [comment]
I'm also seeing this with `--version=v3.6.0` using current `oc` from built from master

```
-- Installing service catalog ...
I0731 11:04:31.353607   19124 servicecatalog.go:93] instantiating service catalog template with parameters map[SERVICE_CATALOG_SERVICE_IP:172.30.1.2 CORS_ALLOWED_ORIGIN:127.0.0.1 SERVICE_CATALOG_IMAGE:openshift/origin-service-catalog:v3.6.0]
FAIL
   Error: cannot instantiate service catalog template
   Caused By:
     Error: cannot create objects from template openshift-infra/service-catalog
     Caused By:
       Error: role.authorization.openshift.io "extension-apiserver-authentication-reader" already exists
```

**Comment by bparees** [comment]
@deads2k why would that role already exist in a 3.6 server?

**Comment by deads2k** [comment]
> @deads2k why would that role already exist in a 3.6 server?

stale etcd?

**Comment by spadgett** [comment]
I was clustering up fresh with no existing data. @bparees do you see the error too with `--version=v3.6.0`?

**Comment by bparees** [comment]
just to follow up, it looks like the 3.6.0 released images are actually 3.7, so that's why the role exists.
